conversion/conversion.go

```diff
@@ -283,14 +289,66 @@ type IOStreams struct {
 	BytesRead           int64
 }
 
+// downloadFromGCS returns the path of a dump file that is downloaded from GCS
+func downloadFromGCS(dumpFile string) (*os.File, error){
+	split := strings.Index(dumpFile, "/")
+	bucketName := dumpFile[:split]
+	filePath := dumpFile[split+1:]
+	fmt.Printf("\nDownloading dump file from GCS bucket %s, path %s\n", bucketName, filePath)
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+	defer client.Close()
+
+	bucket := client.Bucket(bucketName)
+
+	rc, err := bucket.Object(filePath).NewReader(ctx)
+	if err != nil {
+		fmt.Printf("readFile: unable to open file from bucket %q, file %q: %v", bucketName, filePath, err)
+		log.Fatal(err)
+		return nil, err
+	}
+	defer rc.Close()
+
+	data, err := ioutil.ReadAll(rc)
```

Comment:
This will cause the whole file to be loaded into memory before writing it out to file.
It might be a challenge for files of a few GB or more. An appropriate approach is to write to the file as you are reading, so as to reduce the memory requirement of the process